functions/src/timetables/__tests__/Timetable.controller.test.ts

```diff
@@ -0,0 +1,180 @@
+import { mocked } from 'ts-jest/utils';
+import {
+  addTimetable,
+  getTimetable,
+  TimetableParams,
+} from '../Timetable.service';
+
+import * as request from 'supertest';
+import * as express from 'express';
+import { Response as ExResponse, Request as ExRequest } from 'express';
+import { RegisterRoutes } from '../../../build/routes';
+import { ValidateError } from '@tsoa/runtime';
+
+jest.mock('../Timetable.service.ts');
+
+const mockGetTimetable = mocked(getTimetable);
+const mockAddTimetable = mocked(addTimetable);
+
+const app = express();
+
+// Use body parser to read sent json payloads
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+RegisterRoutes(app);
+
+// Need this to test the validation checks
+app.use(function errorHandler(
+  err: unknown,
+  req: ExRequest,
+  res: ExResponse,
+  next: express.NextFunction
+): ExResponse | void {
+  if (err instanceof ValidateError) {
+    console.warn(`Caught Validation Error for ${req.path}:`, err.fields);
+    return res.status(422).json({
+      message: 'Validation Failed',
+      details: err?.fields,
+    });
+  }
+  if (err instanceof Error) {
+    return res.status(500).json({
+      message: 'Internal Server Error',
+    });
+  }
+
+  next();
+});
```

Comment:
nit: can we move this into the middlewares directory since it's used in a bunch of spots.